### PR TITLE
feat(opapi): support for binary content types

### PR DESCRIPTION
**Pull request by spg**
Also adds support for 302 redirects

Closes CLS-1790

### CODE REVIEW

File: opapi/src/state.ts [hunk 1]
```diff
@@ -196,7 +198,7 @@ type BaseOperationProps<
   response: {
     // Status code of the response
     // Default is 200
-    status?: 200 | 201 | 418
+    status?: 200 | 201 | 418 | 302
```
Comment (franklevasseur) [inline opapi/src/state.ts:201]: :+1:

File: opapi/src/state.ts [hunk 2]
```diff
@@ -124,7 +124,9 @@ export type OperationWithBodyProps<
   requestBody: {
     description: string
     schema: SchemaOfType<S>
+    format?: 'binary'
```
Comment (franklevasseur) [inline opapi/src/state.ts:127]: **question:** what does that mean ? I know its an `@anatine/zod-openapi`, but what will be the effect on the resulting openapi ?

Also, can you use:

```ts
{
  requestBody: { format: "binary" },
  contentType: "application/json"
}
```

or is `{ format: "binary" }` absolutely tied with `{ contentType: "*/*" }` ?

**Review by franklevasseur — COMMENTED** [review]
Didn't you need to also support binary content for responses ?

let's have a quick talk before merging

File: opapi/src/state.ts (hunk 2 above)
Comment (spg) [reply opapi/src/state.ts:127]: `format` is tied to a fields of `type: string`. AFAIK, it has nothing to do with the `contentType`.

**Review by franklevasseur — APPROVED** [review]

**Pull request merged**